StitchCore-iOS/StitchCore-iOS/Core/Stitch.swift

```diff
@@ -37,25 +36,26 @@ public class Stitch {
      * - throws: A `StitchInitializationError` if initialization fails for any reason.
      */
     public static func initialize() throws {
-        try! StitchCore.sync(self) {
-            guard !initialized else { return }
+        objc_sync_enter(self)
+        defer { objc_sync_exit(self) }
```

Comment:
What's the difference between having this at the line above the enter?

Reply:
If for whatever reason the code exits the functions before `enter`, then `exit` will be called when it should not have been.

Reply:
That can cause a crash or a hanging lock

Reply:
But the same can be said if, for whatever reason, the code exits the function after `enter` but an `exit` has not been stated. That would also cause a crash or hanging lock.

Reply:
Almost. It would cause a hanging lock. Either way it's unlikely but you never should write code that unlocks something before you even locked it. It's too error prone to do it in the incorrect order such that you create code that goes in the wrong order. Imagine you have the defer but then only conditionally `enter` in some cases. Then you may accidentally hang as opposed to keeping `enter` and `exit` near each other, `defer` or not.

Reply:
I agree with Eric here, so going to leave as is.